plugins/scaffolder/src/components/TemplateCard/TemplateCard.tsx

```diff
@@ -72,7 +86,54 @@ const getTemplateCardProps = (
   };
 };
 
-export const TemplateCard = ({ template }: TemplateCardProps) => {
+const DeprecationWarning = () => {
+  const [anchorEl, setAnchorEl] = React.useState<SVGSVGElement | null>(null);
+
+  const handleClick = (event: React.MouseEvent<SVGSVGElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const styles = useDeprecationStyles();
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const open = Boolean(anchorEl);
+
+  const Title = (
+    <Typography style={{ padding: 10, maxWidth: 300 }}>
+      This template syntax is deprecated.
+      <br />
+      <Link href="https://backstage.io/docs/features/software-templates/migrating-from-v1alpha1-to-v1beta2">
+        Please follow the documentation to migrate
+      </Link>
+    </Typography>
+  );
+
+  return (
+    <div className={styles.deprecationIcon}>
+      <WarningIcon onClick={handleClick} />
+      <Popover
```

Comment:
instead of dealing with the popover manually, Tooltip supports being controlled too https://material-ui.com/components/tooltips/#controlled-tooltips

not sure if it makes things vastly smaller, but just a (tool)tip

Reply:
could be combined with making the icon itself the clickable element, then the tooltip text doesn't have to contain any magic

Reply:
and also doesn't have to be controlled at all with explicit open/close handling - THAT should make this significantly simpler

Reply:
I tried that first but the tooltip does not allow clickable content.

Reply:
or you mean making the icon be the link out?

Reply:
yes

Reply:
just keeping things simple :)